Libs/Core/Testing/Cpp/ctkCoreSettingsTest.cpp

```diff
@@ -0,0 +1,164 @@
+/*=========================================================================
+
+  Library:   CTK
+
+  Copyright (c) Kitware Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0.txt
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+=========================================================================*/
+
+// CTK includes
+#include "ctkCoreSettings.h"
+#include "ctkTest.h"
+
+// STD includes
+#include <iostream>
+#include <limits>
+
+// Qt includes
+#include <QApplication>
+
+// ----------------------------------------------------------------------------
+class ctkCoreSettingsTester: public QObject
+{
+  Q_OBJECT
+private slots:
+  void initTestCase();
+
+  void testSinglePath();
+  void testSinglePath_data();
+
+  void testMultiplePaths();
+  void testMultiplePaths_data();
+};
+
+// ----------------------------------------------------------------------------
+void ctkCoreSettingsTester::initTestCase()
+{
+}
+
+// ----------------------------------------------------------------------------
+void ctkCoreSettingsTester::testSinglePath()
+{
+  QFETCH(QString, homeDir);
+  QFETCH(QString, inputPath);
+  QFETCH(QString, expectedStoredValue);
+
+  ctkCoreSettings settings;
+  settings.setApplicationHomeDirectory(homeDir);
+  QString key("someKey");
+  settings.setPathValue(key, inputPath);
+
+  QCOMPARE(settings.value(key).toString(), expectedStoredValue);
+
+#ifdef Q_OS_WIN
+  QCOMPARE(settings.pathValue(key).toString().toLower(), QDir::cleanPath(inputPath).toLower());
```

Comment:
Since macOS can also have case insensitive filesystem, should we systematically user `toLower()` in the test ?

Reply:
done